src/secret/mnemonic.rs

```diff
@@ -13,96 +16,62 @@ use crypto::{
     keys::slip10::{Chain, Curve, Seed},
 };
 
-use super::{types::InputSigningData, SignMessageMetadata};
-use crate::{
-    constants::HD_WALLET_TYPE,
-    signing::{SignerHandle, SignerType},
-    Client, Result,
-};
-
-fn generate_addresses(
-    seed: &Seed,
-    coin_type: u32,
-    account_index: u32,
-    address_indexes: Range<u32>,
-    internal: bool,
-) -> crate::Result<Vec<Address>> {
-    let mut addresses = Vec::new();
-    for address_index in address_indexes {
-        let chain = Chain::from_u32_hardened(vec![
-            HD_WALLET_TYPE,
-            coin_type,
-            account_index,
-            internal as u32,
-            address_index,
-        ]);
-        let public_key = seed
-            .derive(Curve::Ed25519, &chain)?
-            .secret_key()
-            .public_key()
-            .to_bytes();
-        // Hash the public key to get the address
-        let result = Blake2b256::digest(&public_key)
-            .try_into()
-            .map_err(|_e| crate::Error::Blake2b256Error("Hashing the public key while generating the address failed."));
-
-        addresses.push(Address::Ed25519(Ed25519Address::new(result?)));
-    }
-    Ok(addresses)
-}
-
-/// MnemonicSigner, also used for seeds
-pub struct MnemonicSigner(Seed);
+use super::{types::InputSigningData, GenerateAddressMetadata, SecretManager, SignMessageMetadata};
+use crate::{constants::HD_WALLET_TYPE, Client, Result};
 
-impl Deref for MnemonicSigner {
-    type Target = Seed;
-    fn deref(&self) -> &Self::Target {
-        &self.0
-    }
-}
-
-impl MnemonicSigner {
-    /// Create a new MnemonicSigner SignerHandle with a given BIP39 mnemonic from the English wordlist
-    /// for more information see <https://github.com/bitcoin/bips/blob/master/bip-0039.mediawiki>
-    #[allow(clippy::new_ret_no_self)]
-    pub fn new(mnemonic: &str) -> Result<SignerHandle> {
-        Ok(SignerHandle::new(
-            SignerType::Mnemonic,
-            Box::new(Self(Client::mnemonic_to_seed(mnemonic)?)),
-        ))
-    }
-    /// Create a new MnemonicSigner SignerHandle with a given hex encoded seed
-    pub fn new_from_seed(seed: &str) -> Result<SignerHandle> {
-        Ok(SignerHandle::new(
-            SignerType::Mnemonic,
-            Box::new(Self(Seed::from_bytes(&hex::decode(seed)?))),
-        ))
-    }
-}
+/// Secret manager that uses only a mnemonic.
+///
+/// Computation are done in-memory. A mnemonic needs to be supplied upon the creation of [`MnemonicSecretManager`].
+pub struct MnemonicSecretManager(Seed);
```

Comment:
```suggestion
#[derive(Clone)]
pub struct MnemonicSecretManager(Seed);
```

Reply:
`Seed` is not `Clone`

Reply:
Oh right, we only had clone for `SignerHandle` because it had an Arc, so all good how it's now